Content.Server/GameObjects/Components/Atmos/AirtightComponent.cs

```diff
@@ -111,13 +156,18 @@ private void OnTransformMove()
 
         private void UpdatePosition()
         {
-            var mapIndices = Owner.Transform.Coordinates.ToMapIndices(_entityManager, _mapManager);
-            UpdatePosition(Owner.Transform.GridID, mapIndices);
+            if (Owner.TryGetComponent(out SnapGridComponent? snapGrid))
+                UpdatePosition(Owner.Transform.GridID, snapGrid.Position);
         }
 
         private void UpdatePosition(GridId gridId, MapIndices pos)
         {
-            EntitySystem.Get<AtmosphereSystem>().GetGridAtmosphere(gridId)?.Invalidate(pos);
+            var gridAtmos = EntitySystem.Get<AtmosphereSystem>().GetGridAtmosphere(gridId);
```

Comment:
The entity system here could be cached if you are concerned about the amount of times this method is going to be called.

Reply:
done